plugin/src/main/java/org/opensearch/ml/cluster/DiscoveryNodeHelper.java

```diff
@@ -67,15 +66,17 @@ public DiscoveryNode[] getEligibleNodes() {
                 eligibleDataNodes.add(node);
             }
         }
-        if (eligibleMLNodes.size() > 0) {
-            DiscoveryNode[] mlNodes = eligibleMLNodes.toArray(new DiscoveryNode[0]);
-            log.debug("Find {} dedicated ML nodes: {}", eligibleMLNodes.size(), Arrays.toString(mlNodes));
-            return mlNodes;
-        } else {
-            DiscoveryNode[] dataNodes = eligibleDataNodes.toArray(new DiscoveryNode[0]);
-            log.debug("Find no dedicated ML nodes. But have {} data nodes: {}", eligibleDataNodes.size(), Arrays.toString(dataNodes));
-            return dataNodes;
-        }
+        eligibleMLNodes.addAll(eligibleDataNodes);
+        return eligibleMLNodes.toArray(new DiscoveryNode[0]);
```

Comment:
This is breaking the original design. In the original design, if ML node exists, we always use ML node to deploy model, no matter run on ML node only setting enabled or disabled. That can help protect the cluster as running model on data node could crash the data nodes and bring the whole cluster down.

This is a known issue: If all model deployed on data nodes, then later add new ML node, the cron job will change planning worker node to ML node, but worker node still shows the data nodes. And it's possible worker node count is greater than planning worker node count. That's why the code is using planning worker node `>` worker node, rather than "=".   
Before model auto-reload, we plan to ask user to do manually undeploy, then re-deploy.
As we are close to code freeze date for OS2.7, I'm ok to skip this case in auto-reload feature, we can enhance later.

Reply:
Sure, code reverted.